### ISSUE TITLE
`namespace-as-table?` option is being applied on INSERT INTO field list when it shouldn't.

### ISSUE THREAD

**Issue by jrdoane** [jkk/honeysql#259]
When using the `namespace-as-table?` option to format SQL, fully qualified names are being put in places where fully qualified names are not valid. For example, using a fully qualified name in an `INSERT INTO` statement for the field names is not valid and the SQL-92 spec specifically calls out that these column names should be unqualified. The real-world implication of this is that without working around the issue, queries produced this way will fail when executed against a PostgreSQL database.

So when running something like this:
```clojure
(sql/format
  {:insert-into :potato
   :values [{:potato/id 1
             :potato/value 2}]}
  :namespace-as-table? true 
  :quoting :ansi
  :allow-dashed-names? true)
```
Results in this:

```clojure
["INSERT INTO \"potato\" (\"potato\".\"id\", \"potato\".\"value\") VALUES (?, ?)" 1 2]
```

...and not this:

```clojure
["INSERT INTO \"potato\" (\"id\", \"value\") VALUES (?, ?)" 1 2]
```

The workaround is pretty obvious, don't pass the `:namespace-as-table?` option when doing `INSERT INTO` statements, but this means that applications using HoneySQL will need to manage toggling this flag on and off based on the type of query that's being executed and doesn't make for as pleasant of an experience as it could be.

Resources:
[SQL-92 Spec](https://pubs.opengroup.org/onlinepubs/9695959099/toc.pdf) (Pages 119-120)

**Comment by seancorfield** [comment]
Interesting. Some databases allow dotted names there so I haven't tripped over this issue with the databases I use. I'll add some tests around this, across multiple databases, and then figure out how best to solve the problem.

**Comment by jrdoane** [comment]
I encountered this with a new project I started and part of my plan was to use `next.jdbc` instead of `clojure.java.jdbc` and I wanted to maintain the namespaced keywords across the board so I could write consistent specs in the future. That's how I encountered this bug.

**Comment by seancorfield** [comment]
I updated the tests in `next.jdbc` to see which DBs accepted it and which didn't -- it's not an exhaustive list:

- H2, HSQLDB, SQLite, PostgreSQL -- do not accept it
- MySQL, MS SQL Server -- accept dotted names

So it's pretty clear that HoneySQL needs to ensure the identifiers clause of `insert` is not namespaced.

**Comment by seancorfield** [comment]
Fixed on **master** in d86150784bd81c7b881727e89eaa5b10904022e3
